Api/Services/RestaurantService.cs

```diff
@@ -428,6 +428,256 @@ public async Task<Result<List<RestaurantEmployeeVM>>> GetEmployeesAsync(int id,
+                        ? null : uploadService.GetPathForFileName(r.RentalContractFileName),
+                    AlcoholLicense = r.AlcoholLicenseFileName == null
+                        ? null : uploadService.GetPathForFileName(r.AlcoholLicenseFileName),
+                    BusinessPermission = uploadService.GetPathForFileName(r.BusinessPermissionFileName),
+                    IdCard = uploadService.GetPathForFileName(r.IdCardFileName),
+                    Tables = r.Tables!.Select(t => new TableVM
+                    {
+                        Id = t.Id,
+                        Capacity = t.Capacity
+                    }),
+                    Photos = r.Photos!
+                        .OrderBy(rp => rp.Order)
+                        .Select(rp => uploadService.GetPathForFileName(rp.PhotoFileName))
+                        .ToList(),
+                    ProvideDelivery = r.ProvideDelivery,
+                    Logo = uploadService.GetPathForFileName(r.LogoFileName),
+                    Description = r.Description,
+                    Tags = r.Tags!.Select(t => t.Name).ToList()
+                })
+                .AsSplitQuery()
+                .FirstOrDefaultAsync();
+
+            return newRestaurant;
+        }
+
+        /// <summary>
+        /// Deletes a restaurant
+        /// </summary>
+        /// <param name="id">ID of the restaurant to delete</param>
+        /// <param name="user">User requesting a deletion</param>
+        public async Task<Result<bool>> DeleteRestaurantAsync(int id, User user)
+        {
+            var errors = new List<ValidationResult>();
+    
+            var restaurant = await context.Restaurants
+                .Include(r => r.Group)
+                .Include(r => r.Photos)
+                .Include(r => r.Tags)
+                .Include(r => r.Tables)
+                .Include(r => r.Employments)
+                .FirstOrDefaultAsync(r => r.Id == id);
+
+            if (restaurant == null)
+            {
+                errors.Add(new ValidationResult($"Restaurant with ID {id} not found."));
+                return errors;
+            }
+
+            if (restaurant.Group?.OwnerId != user.Id)
+            {
+                errors.Add(new ValidationResult("User is not the owner of this restaurant."));
+                return errors;
+            }
+
+            // Removing dependencies
+            context.RestaurantPhotos.RemoveRange(restaurant.Photos);
+            context.Tables.RemoveRange(restaurant.Tables);
+            context.Employments.RemoveRange(restaurant.Employments);
+
+
+            context.Restaurants.Remove(restaurant);
+
+            try
+            {
+                await context.SaveChangesAsync();
+                return true;
+            }
+            catch (Exception ex)
+            {
+                errors.Add(new ValidationResult("Failed to delete restaurant: " + ex.Message));
+                return errors;
+            }
+        }
```

Comment:
Nie miałeś robić DELETE, bo to miał Michał G (a i tak robimy soft delete więc nic nie usuwamy pod spodem)